editoast/src/views/projects.rs

```diff
@@ -267,13 +250,35 @@ async fn patch(
 ) -> Result<Json<ProjectWithStudies>> {
     let data = data.into_inner();
     let project_id = project.into_inner();
-    let project = data.into_project(project_id);
-    if let Some(Some(image)) = project.image {
+    if let Some(image) = data.image {
         check_image_content(db_pool.clone(), image).await?;
     }
-    let project = match project.update(db_pool.clone()).await? {
-        Some(project) => project,
-        None => return Err(ProjectError::NotFound { project_id }.into()),
+    let mut project_changeset = Project::changeset();
+    if let Some(name) = data.name {
+        project_changeset = project_changeset.name(name);
+    }
+    if let Some(description) = data.description {
+        project_changeset = project_changeset.description(description);
+    }
+    if let Some(objectives) = data.objectives {
+        project_changeset = project_changeset.objectives(objectives);
+    }
+    if let Some(funders) = data.funders {
+        project_changeset = project_changeset.funders(funders);
+    }
+    if let Some(budget) = data.budget {
+        project_changeset = project_changeset.budget(budget);
+    }
+    if let Some(image) = data.image {
+        project_changeset = project_changeset.image(Some(image));
+    }
+    if let Some(tags) = data.tags {
+        project_changeset = project_changeset.tags(tags);
+    }
```

Comment:
```suggestion
let project_changeset = ProjectChangeset {
    name: data.name,
    objectives: data.objectives,
    ....
}
```

Reply:
I can't create ProjecteChangest like this, all field are private.

Reply:
Hmmm, I missed that use case. I'll update the builder to allow giving Option<>. In the meantime you can use `#[model(changeset(public))]`.